include/jemalloc/internal/arena_inlines_b.h

```diff
@@ -298,6 +298,56 @@ arena_dalloc_large(tsdn_t *tsdn, void *ptr, tcache_t *tcache, szind_t szind,
 	}
 }
 
+/* Find the region index of a pointer. */
+JEMALLOC_ALWAYS_INLINE size_t
+arena_slab_regind_impl(div_info_t* div_info, szind_t binind,
+    edata_t *slab, const void *ptr) {
+	size_t diff, regind;
+
+	/* Freeing a pointer outside the slab can cause assertion failure. */
+	assert((uintptr_t)ptr >= (uintptr_t)edata_addr_get(slab));
+	assert((uintptr_t)ptr < (uintptr_t)edata_past_get(slab));
+	/* Freeing an interior pointer can cause assertion failure. */
+	assert(((uintptr_t)ptr - (uintptr_t)edata_addr_get(slab)) %
+	    (uintptr_t)bin_infos[binind].reg_size == 0);
+
+	diff = (size_t)((uintptr_t)ptr - (uintptr_t)edata_addr_get(slab));
+
+	/* Avoid doing division with a variable divisor. */
+	regind = div_compute(div_info, diff);
+	assert(regind < bin_infos[binind].nregs);
+	return regind;
+}
+
+/*
+* Checks whether ptr is already deallocated in the arena.
+* Fires safety check if it has been deallocated and returns true.
+*/
```

Comment:
nit: the format -- stars should all align (like the section you have below).

Also probably remove the second line -- the "returns true" part can cause confusion, since in actual workload / non unit tests, the safety check failure never returns to that point at all.

Reply:
indeed, modified and the second line removed, since it may simply be a corrupted pointer instead of already deallocated, modified the comment as well.